index.bs

```diff
@@ -575,79 +575,225 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+
+			1. Let |types| be a list of {{DOMString}}.
+
+			1. For each (|key|, |value|) in |items|:
+
+				1. Let |mimeType| be the result of [=parse a mime type=] given |key|.
+
+				1. If |mimeType| is failure, then throw a {{TypeError}}.
+
+				1. Let |representation| be a new [=representation=].
+
+				1. Set |representation|'s [=mime type=] to |key|.
+
+				1. Set |representation|'s [=data=] to |value|.
+
+				1. Append |representation| to [=this=]'s [=ClipboardItem/clipboard item=]'s [=list of representations=].
+
+				1. Let |mimeTypeString| be the result of [=serialize a mime type=] with |mimeType|.
+				
+				1. Add |mimeTypeString| to |types|.
+
+			1. Set [=this=]'s [=ClipboardItem/types array=] to the result of running [=create a frozen array=] from |types|.
+
+		<h4 attribute for=ClipboardItem lt=presentationStyle>presentationStyle</h4>
+		<p>
+			The {{ClipboardItem/presentationStyle}} getter steps are to return [=this=]'s [=ClipboardItem/clipboard item=]'s [=presentation style=].
+		</p>
+
+		<h4 attribute for=ClipboardItem lt=types>types</h4>
+		<p>
+			{{ClipboardItem/types}} getter steps are to return [=this=]'s [=ClipboardItem/types array=].
+		</p>
+
+		<h4 method for=ClipboardItem lt=getType(type)>getType(type)</a> must run the below steps:</h4>
+
+			1. Let |realm| be [=this=]'s [=relevant realm=].
+
+			1. Let |mimeType| be the result of [=parse a mime type=] given |type|.
+
+			1. If |mimeType| is failure, then throw a {{TypeError}}.
+
+			1. Let |itemTypeList| be [=this=]'s [=ClipboardItem/clipboard item=]'s [=list of representations=].
+
+			1. For each |representation| in |itemTypeList|:
+
+				1. If |representation|'s [=mime type=] is |mimeType|, then:
+
+					1. Let |blobData| be a [=Blob=] created using |representation|'s [=data=] and with its {{Blob/type}} set to |mimeType|, [=serialize a MIME type|serialized=].
 
-		enum PresentationStyle { "unspecified", "inline", "attachment" };
+					1. Let |p| be [=a new promise=] in |realm|.
 
-		dictionary ClipboardItemOptions {
-		  PresentationStyle presentationStyle = "unspecified";
+					1. Resolve |p| with |blobData|.
+
+					1. Return |p|.
+
+			1. Return [=a promise rejected with=] {{"NotFoundError"}} DOMException in |realm|.
+
+	<h3 id="clipboard-interface">Clipboard Interface</h3>
+
+		<pre class="idl" data-highlight="webidl">
+		typedef sequence&lt;ClipboardItem&gt; ClipboardItems;
+
+		[SecureContext, Exposed=Window]
+		interface Clipboard : EventTarget {
+			Promise&lt;ClipboardItems&gt; read();
+			Promise&lt;DOMString&gt; readText();
+			Promise&lt;undefined&gt; write(ClipboardItems data);
+			Promise&lt;undefined&gt; writeText(DOMString data);
 		};
 		</pre>
 
-		<div id="clipboard-idl" dfn-for="Clipboard">
+		A <dfn>Clipboard</dfn> may contain <dfn>Clipboard Items</dfn> which has one or more [=clipboard item=]s. Multiple [=clipboard item=]s are only supported on platforms such as iOS/iPadOS.
```

Comment:
As mentioned in [this comment](https://github.com/w3c/clipboard-apis/pull/158/#discussion_r766517505), it seems the term "Clipboard Items" shouldn't be defined. The marked line contains a `<dfn>Clipboard Items<dfn>` which defines the term "Clipboard Items" without precisely defining it.

The `<dfn>Clipboard<dfn>` in the same line redefines the Clipboard defined via IDL; it should be a reference instead, i.e. `<code><a ...>Clipboard</a></code>`. Moreover, the `Clipboard` interface doesn't "have" clipboard items. It operates with them.
So something like

`A <code><a ...>Clipboard</a></code>'s <code><a ...>read()</a></code> and <code><a ...>write()</a></code> method must support multiple <code><a ...>ClipboardItem</a></code>s, when the operating systems supports them. <p class="note" ...>Currently, only operating systems from Apple (e.g. iOS) support multiple <code><a ...>ClipboardItem</a></code>s` seems more correct. WDYT?

Reply:
Yeah, I think you have raised a good point here. `Clipboard` doesn't contain `Clipboard Items`. It just uses `Clipboard Items` in the `read()` & `write()`. Not sure if there is a better way to write this, but I can make this change if @domenic agrees with it as well.